packages/salesforcedx-vscode-core/test/jest/orgPicker/orgList.test.ts

```diff
@@ -0,0 +1,334 @@
+  let getAllMock: jest.SpyInstance;
+  let getAllAliasesForMock: jest.SpyInstance;
+  let getUsernameForMock: jest.SpyInstance;
+  let getAuthFieldsForMock: jest.SpyInstance;
+  let stateAggregatorCreateMock: jest.SpyInstance;
+  let fakeStateAggregator: any;
+  const mockStatusBarItem: any = {};
+  let mockWatcher: any;
+
+  const dummyDevHubUsername = 'test-devhub@example.com';
+
+  const createOrgAuthorization = (
+    overrides: Partial<OrgAuthorization> = {}
+  ): OrgAuthorization => ({
+    orgId: '000',
+    username: 'test-username@example.com',
+    oauthMethod: 'unknown',
+    aliases: [],
+    configs: [],
+    isScratchOrg: undefined,
+    isDevHub: undefined,
+    isSandbox: undefined,
+    instanceUrl: undefined,
+    accessToken: undefined,
+    error: undefined,
+    isExpired: false,
+    ...overrides
+  });
+
+  const orgAuth = createOrgAuthorization();
+  const orgAuthWithError = createOrgAuthorization({
+    username: 'error-org@example.com',
+    error: 'Some error'
+  });
+  const orgAuthScratchOrg = createOrgAuthorization({ isScratchOrg: true });
+  const orgAuthWithAlias = createOrgAuthorization({
+    username: 'test-username@example.com',
+    aliases: ['alias1']
+  });
+  const validOrgAuth = createOrgAuthorization({
+    username: 'valid-org@example.com'
+  });
+
+  beforeEach(() => {
+    mockStatusBarItem.tooltip = '';
+    mockStatusBarItem.command = '';
+    mockStatusBarItem.text = '';
+    mockStatusBarItem.show = jest.fn();
+    mockStatusBarItem.dispose = jest.fn();
+    mockWatcher = {
+      onDidChange: jest.fn(),
+      onDidCreate: jest.fn(),
+      onDidDelete: jest.fn()
+    };
+    createFileSystemWatcherMock = (
+      vscode.workspace.createFileSystemWatcher as any
+    ).mockReturnValue(mockWatcher);
+    createStatusBarItemMock = (
+      vscode.window.createStatusBarItem as jest.Mock
+    ).mockReturnValue(mockStatusBarItem);
+    orgList = new OrgList();
+    getAuthFieldsForMock = jest.spyOn(OrgList.prototype, 'getAuthFieldsFor');
+    getUsernameForMock = jest.spyOn(ConfigUtil, 'getUsernameFor');
+    getDevHubUsernameMock = jest.spyOn(OrgAuthInfo, 'getDevHubUsername');
+    getAllMock = jest.fn();
+    fakeStateAggregator = {
+      aliases: {
+        getAll: getAllMock
+      }
+    };
+    stateAggregatorCreateMock = jest
+      .spyOn(StateAggregator, 'create')
+      .mockResolvedValue(fakeStateAggregator);
+    getAllAliasesForMock = jest.spyOn(ConfigUtil, 'getAllAliasesFor');
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('filterAuthInfo', () => {
+    it('should filter out orgs with the scratchAdminUsername field', async () => {
+      const orgAuths = [orgAuth];
+      getAuthFieldsForMock.mockResolvedValueOnce({
+        scratchAdminUsername: 'admin@example.com'
+      } as AuthFields);
+      getDevHubUsernameMock.mockResolvedValueOnce(null);
+
+      const result = await orgList.filterAuthInfo(orgAuths);
+
+      expect(result).toEqual([]);
+    });
```

Comment:
Why do we want to filter out scratch orgs with the `scratchAdminUsername` field?  What's special about them?

Reply:
I don't know, most of these tests were in mocha and I just translated them to jest